rbx_reflector/src/patches.rs

```diff
@@ -90,6 +91,65 @@ impl Patches {
 
         Ok(())
     }
+
+    pub fn apply_post_default(&self, database: &mut ReflectionDatabase) -> anyhow::Result<()> {
+        // A map of every class to all subclasses, by name. This uses `String`
+        // rather than some borrowed variant to get around borrowing `database`
+        // as both mutable and immutable
+        let mut subclass_map: HashMap<String, Vec<String>> =
+            HashMap::with_capacity(database.classes.len());
+
+        for (class_name, class_descriptor) in &database.classes {
+            for superclass in database.superclasses(class_descriptor).unwrap() {
+                subclass_map
+                    .entry(superclass.name.to_string())
+                    .or_default()
+                    .push(class_name.to_string());
+            }
+        }
+
+        for (class_name, class_changes) in &self.change {
+            for (prop_name, prop_change) in class_changes {
+                let default_value = match &prop_change.default_value {
+                    Some(value) => value,
+                    None => continue,
+                };
+                let prop_data = database
+                    .classes
+                    .get(class_name.as_str())
+                    // This is already validated pre-default application, so unwrap is fine
+                    .unwrap()
+                    .properties
+                    .get(prop_name.as_str());
+                if let Some(prop_data) = prop_data {
+                    match (&prop_data.data_type, default_value.ty()) {
+                        (DataType::Enum(_), VariantType::Enum) => {},
+                        (DataType::Value(existing), new) if *existing == new => {},
+                        (existing, new) => bail!(
+                            "Property {class_name}.{prop_name}'s value type is being changed by a DefaultValue patch.\n\
+                            Existing: {existing:?}, new: {new:?}")
```

Comment:
This is a bit unclear to me, I'd prefer something like
```suggestion
                        (expected, actual) => bail!(
                            "Bad type given for {class_name}.{prop_name}'s DefaultValue patch.\n\
                            Expected {expected:?}, got {actual:?}")
```